### DISCUSSION TITLE
feature request : MSI Paket build for Automatic agent deployment

### DISCUSSION THREAD

**Discussion by telenoise** [Ylianst/MeshCentral#5051] [Ideas]
We are wrapping the agent.exe file into a msi to deploy this via Active Director.... it might be a good idea to have this downloadable...
any ideas on this ?

**Comment by si458** [comment]
You can download the exe, then include it in an msi and run `theagent.exe -fullinstall`, and this will silently install it

But yes, an msi for silent install would be a good idea

**Comment by ke6yjc** [comment]
What about trying to use a generic agent installer and telling it which device group to join? What we have done for now is we made a device group calling on-boarding and then we deploy the exe with -fullinstall but it would be nice if there was a way to not have to build an exe for each device group and just reference a guid/groupid, etc. Thanks in Advance!

**Reply by telenoise** [reply]
yes nice idea

**Reply by si458** [reply]
@ke6yjc @telenoise something is already possible as such.
Simply download the meshagent from the meshcentral-data/agents folder (not signedagents)
Then if u add `?debug=1` into your address bar, when u goto download the meshagent and a new option is there to download the .msh file
Download this file for each meshgroup and u will see whats different
Then u can build ur own msi to deploy the .msh and exe somewhere then run the normal --fullinstall or even just --install
So u can sorta automate your deployments!
